src/command/process_update_task.rs

```diff
@@ -344,6 +508,68 @@ impl UpdateProcessor {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, TypedBuilder, Eq, PartialEq, Clone)]
+#[serde(crate = "fang::serde")]
+pub struct ScheduleWeatherTask {
+    user_hour: i8,
+    minutes: i8,
+    offset: i8,
+    username: String,
+    chat_id: i64,
+    default_city_id: i32,
+}
+
+#[typetag::serde]
+#[async_trait]
+impl AsyncRunnable for ScheduleWeatherTask {
+    async fn run(&self, _queueable: &mut dyn AsyncQueueable) -> Result<(), FangError> {
+        // here we should program the weather_info deliver
+        let repo = Repo::repo().await.unwrap();
+        let api = ApiClient::api_client().await;
+
+        let city = repo.search_city_by_id(&self.default_city_id).await.unwrap();
+
+        let weather_client = WeatherApiClient::weather_client().await;
+
+        let weather_info = weather_client
+            .fetch_weekly(city.coord.lat, city.coord.lon)
+            .await
+            .unwrap();
+
+        let text = format!(
+            "Hi {} !, this is your scheduled weather info.\n\n {},{}\nLat {} , Lon {}\n{}",
+            self.username, city.name, city.country, city.coord.lat, city.coord.lon, weather_info,
+        );
+
+        api.send_message_without_reply(self.chat_id, text)
+            .await
+            .unwrap();
+
+        Ok(())
+    }
+
+    fn uniq(&self) -> bool {
+        true
+    }
+
+    fn task_type(&self) -> String {
+        TASK_TYPE.to_string()
```

Comment:
I think you can start separate workers to process `ScheduleWeatherTask`

Reply:
[Scheduled Forecast workers separated from update processor workers](https://github.com/pxp9/weather_bot_rust/pull/29/commits/0d8880add7d568380b4d5aa692fa5fcf8be27cc1)